### DISCUSSION TITLE
Triggering a Power Automate Flow when a custom API is called

### DISCUSSION THREAD

**Discussion by huseyinazizcan** [drivardxrm/XTB.CustomApiManager#27] [Q&A] [answered]
Hi,

Using your tool and the guide below, I created a custom API that triggers a Power Automate flow with the "When an action is performed" Dataverse trigger. My custom API has a single output parameter called result (string).

My question is: Is there a way for the Power Automate flow to respond back to the custom API and set this result parameter, so that the API response includes data retrieved via Power Automate?

**Link to guide:**
https://dreamingincrm.com/2021/06/20/using-custom-api-as-a-trigger-for-flow

Thanks

**Comment by drivardxrm** [comment, marked as answer]
Hi @huseyinazizcan,

Thanks for your interest in my tool.
Unfortunately, the scenario you are describing is not possible.

When using the 'When an action is performed' trigger in power Atomate, the underlying Custom API only serves a signal to trigger the workflow and further processing is done asynchronously. So there is no way to send info back to the calling Custom Api instance.

To achieve what you want, I would suggest that you develop the Custom API logic through a Dataverse Plugin. That way you can be synchronous and populate the output accordinly.

Hope this helps,

**Reply by huseyinazizcan** [reply]
Thank you David. I appreciate it.
